pkg/daemon/plugin.go

```diff
@@ -2,60 +2,86 @@ package daemon
 
 import (
 	"fmt"
-	"plugin"
 
 	"github.com/golang/glog"
+
 	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	plugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins"
+	genericplugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins/generic"
+	intelplugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins/intel"
+	k8splugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins/k8s"
+	mellanoxplugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins/mellanox"
+	virtualplugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins/virtual"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
 )
 
-type VendorPlugin interface {
-	// Return the name of plugin
-	Name() string
-	// Return the SpecVersion followed by plugin
-	Spec() string
-	// Invoked when SriovNetworkNodeState CR is created, return if need dain and/or reboot node
-	OnNodeStateAdd(state *sriovnetworkv1.SriovNetworkNodeState) (bool, bool, error)
-	// Invoked when SriovNetworkNodeState CR is updated, return if need dain and/or reboot node
-	OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkNodeState) (bool, bool, error)
-	// Apply config change
-	Apply() error
-}
-
-var pluginMap = map[string]string{
-	"8086": "intel_plugin",
-	"15b3": "mellanox_plugin",
+var VendorPluginMap = map[string]func() (plugin.VendorPlugin, error){
+	"8086": intelplugin.NewPlugin,
+	"15b3": mellanoxplugin.NewMellxanoxPlugin,
 }
 
-const (
-	SpecVersion   = "1.0"
-	GenericPlugin = "generic_plugin"
-	VirtualPlugin = "virtual_plugin"
-	K8sPlugin     = "k8s_plugin"
+var (
+	GenericPlugin     = genericplugin.NewGenericPlugin
+	GenericPluginName = genericplugin.PluginName
+	VirtualPlugin     = virtualplugin.NewVirtualPlugin
+	K8sPlugin         = k8splugin.NewK8sPlugin
 )
 
-// loadPlugin loads a single plugin from a file path
-func loadPlugin(path string) (VendorPlugin, error) {
-	glog.Infof("loadPlugin(): load plugin from %s", path)
-	plug, err := plugin.Open(path)
-	if err != nil {
-		return nil, err
-	}
+func EnablePlugins(platform utils.PlatformType, ns *sriovnetworkv1.SriovNetworkNodeState) (map[string]plugin.VendorPlugin, error) {
```

Comment:
nit : make this package internal ? (`enablePlugins`)

Reply:
done